.azure-pipelines/ultimate-pipeline.yml

```
@@ -1105,6 +1108,205 @@ stages:
+      jobs: [Win]
+
+  - job: Win
+    pool:
+      vmImage: windows-2019
+    timeoutInMinutes: 60 #default value
+    strategy:
+      matrix:
+        x86:
+          platform: x86
+        x64:
+          platform: x64
+    variables:
+      USE_NATIVE_LOADER: true
+    
+    # Enable the Datadog Agent service for this job
+    services:
+      dd_agent: dd_agent
+
+    steps:
+    - template: steps/install-dotnet-sdks.yml
+
+    - task: DownloadPipelineArtifact@2
+      displayName: Download windows profiler home
+      inputs:
+        artifact: windows-profiler-binaries
+        path: $(System.DefaultWorkingDirectory)/profiler/_build
+
+    - template: steps/restore-working-directory.yml
+      parameters:
+        artifact: build-windows-working-directory
+
+    - task: MSBuild@1
+      inputs:
+        solution: profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/Datadog.Profiler.Native.Windows.WithTests.proj
+        configuration: Release
+        platfom: $(platform)
+        msbuildArguments: /target:BuildCppWithAsan /property:DeployToMonitoringHome=True
+      displayName: Build Native profiler and tests
+
+    - powershell: |        
+        dotnet restore /p:Configuration="Release" /p:Platform="$(platform)" profiler/src/Demos/Datadog.Demos.sln
+      displayName: restore sample applications
+
+    - powershell: |        
+        dotnet build --no-restore -c Release -p:Platform=$(platform) profiler/src/Demos/Datadog.Demos.sln
+      displayName: build sample applications
+
+    - powershell: |
+        # Prepare environment
+        $env:DD_INSIDE_CI=$true
+        $env:DD_HOSTNAME="dd-profiler-dotnet"
+        $env:DD_SERVICE="dd-profiler-dotnet"
+        $env:DD_PROFILING_ENABLED="1"
+
+        # Set profiler deployment and test output dir folder path for the test
+        $test_output_dir="$(System.DefaultWorkingDirectory)\test_output"
+
+        $env:DD_TESTING_OUPUT_DIR=$test_output_dir
+        $env:DD_TRACE_DEBUG="1"
+        $env:DD_PROFILING_LOG_DIR="$test_output_dir\logs"
+        $env:DD_PROFILING_OUTPUT_DIR="$test_output_dir\pprofs"
+        $env:DD_TRACE_ENABLED="0"
+        $env:COR_ENABLE_PROFILING="1"
+        $env:COR_PROFILER="{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"
+        $env:COR_PROFILER_PATH="$(Join-Path -Path $(Resolve-Path .) -ChildPath shared/bin/monitoring-home/win-$(platform)/Datadog.Trace.ClrProfiler.Native.dll)"
+        $env:DD_DOTNET_PROFILER_HOME="$(Join-Path -Path $(Resolve-Path .) -ChildPath shared/bin/monitoring-home)"
+
+        # Run Computer01 for 2min
+        cd profiler\_build\bin\Release-$(platform)\profiler\src\Demos\Samples.Computer01\net45
+        .\Samples.Computer01.exe --scenario 1 --timeout 120
+      displayName: run asan test
+      workingDirectory: $(System.DefaultWorkingDirectory)
+
+    - publish: $(System.DefaultWorkingDirectory)/test_output
+      displayName: Uploading test results
+      artifact: asan_tests_windows_profiler_$(platform)_$(System.JobAttempt)
```

can we use the (new) standard format?

```suggestion
      artifact: _$(System.StageName)_$(Agent.JobName)_tests_$(System.JobAttempt)
```

Also, we should probably upload the logs from the logs directory?

---

Oh, I updated all jobs, but missed that one 😢 
Updated here 99a9891dc0c714972dfd5ab19ecbeda191867aa9

As for log directory, I don't have strong opinion. @gleocadie?

---

Ok to upload the logs from logs directory. On Github Actions a faced permission  issues at some point and to avoid all that headache I went for my own repo where I have the permissions.